spec/requests/api/v0/rg_post_article_spec.rb

```diff
@@ -0,0 +1,45 @@
+RSpec.describe Api::V0::ArticlesController, type: :request do
+
+  describe 'research group POST article' do 
+    let(:rg_user) { FactoryBot.create(:user, role: :research_group) }
+    let(:credentials) { rg_user.create_new_auth_token }
+    let(:headers) { {HTTP_ACCEPT: "application/json"}.merge!(credentials) }
+    
+    before do
+      post '/api/v0/articles' , params: {
+        article: {
+          title: 'Test article',
+          body: 'Lorum lorum lorum',
+          author: rg_user.name
+        }
+      }, headers: headers
+    end
+
+    it 'research group user creates new article' do  
+      expect(response.status).to eq 200
+    end
+
+  end
+
+  describe 'university group POST article' do 
+    let(:uni_user) { FactoryBot.create(:user, role: :university) }
+    let(:credentials) { uni_user.create_new_auth_token }
+    let(:headers) { {HTTP_ACCEPT: "application/json"}.merge!(credentials) }
+    
+    before do
+      post '/api/v0/articles' , params: {
+        article: {
+          title: 'Test article',
+          body: 'Lorum lorum lorum',
+          author: uni_user.name
+        }
+      }, headers: headers
+    end
```

Comment:
Please be consistent with university group beeing a reach group

Reply:
This is a sad path test, it's supposed to be university, we are testing that a User with the university role can't create an article

Reply:
What is 'create_new_auth_token'? And what's happening with the var 'credentials' in line 27?

Reply:
we are creating an user and then using the devise token auth to create a token to log that user in, since with the new associations the articles now need to be created by a logged in user

Reply:
@sethnejame please check the Cooper Challenge for that question